### ISSUE TITLE
Issue analyzing text after successful setup

### ISSUE THREAD

**Issue by legendactivated** [GoTeamEpsilon/ctakes-rest-service#43]
Hey Matthew!

Thank you for updating the Readme instructions. It is clear and very very very helpful. I was able to follow the instructions and start the tomcat server successfully. However, when I try to analyze some text, I get below error:

```
22 Sep 2018 12:57:07  INFO DispatcherServlet - FrameworkServlet 'ctakes-rest-service': initialization started
22 Sep 2018 12:57:07  INFO XmlWebApplicationContext - Refreshing WebApplicationContext for namespace 'ctakes-rest-service-servlet': startup date [Sat Sep 22 12:57:07 PDT 2018]; root of context hierarchy
22 Sep 2018 12:57:07  INFO XmlBeanDefinitionReader - Loading XML bean definitions from ServletContext resource [/WEB-INF/ctakes-rest-service-servlet.xml]
22 Sep 2018 12:57:07  INFO RequestMappingHandlerMapping - Mapped "{[/analyze],methods=[POST]}" onto public java.util.Map<java.lang.String, java.util.Map<java.lang.String, java.util.List<java.lang.String>>> org.apache.ctakes.rest.service.CtakesRestController.getAnalyzedJSON(java.lang.String,java.util.Optional<java.lang.String>) throws java.lang.Exception
22 Sep 2018 12:57:07  INFO RequestMappingHandlerAdapter - Looking for @ControllerAdvice: WebApplicationContext for namespace 'ctakes-rest-service-servlet': startup date [Sat Sep 22 12:57:07 PDT 2018]; root of context hierarchy
22 Sep 2018 12:57:07  INFO RequestMappingHandlerAdapter - Looking for @ControllerAdvice: WebApplicationContext for namespace 'ctakes-rest-service-servlet': startup date [Sat Sep 22 12:57:07 PDT 2018]; root of context hierarchy
22 Sep 2018 12:57:07  INFO CtakesRestController - Initializing analysis engines and jcas pools
22 Sep 2018 12:57:07 ERROR PiperFileReader - Unknown Piper Command: threads
22 Sep 2018 12:57:07 ERROR PiperFileReader - Piper File not found: pipers/TsDefaultTokenizerPipeline
22 Sep 2018 12:57:08 ERROR PiperFileReader - Piper File not found: pipers/TsChunkerSubPipe
22 Sep 2018 12:57:08 ERROR PiperFileReader - Piper File not found: pipers/TsDictionarySubPipe
22 Sep 2018 12:57:08 ERROR PiperFileReader - Piper File not found: pipers/TsAttributeCleartkSubPipe
22 Sep 2018 12:57:08  INFO ContextDependentTokenizerAnnotator - Finite state machines loaded.
22 Sep 2018 12:57:08  INFO POSTagger - POS tagger model file: org/apache/ctakes/postagger/models/mayo-pos.zip
22 Sep 2018 12:57:09  WARN PiperFileReader - Attempted to set Parameter by Command-line options.  Command-line options are not specified.
22 Sep 2018 12:57:09  WARN PropertyAeFactory - No parameters specified.
22 Sep 2018 12:57:09  WARN PiperFileReader - Attempted to set Parameter by Command-line options.  Command-line options are not specified.
22 Sep 2018 12:57:09  WARN PropertyAeFactory - No parameters specified.
22 Sep 2018 12:57:09  WARN PiperFileReader - Attempted to set Parameter by Command-line options.  Command-line options are not specified.
22 Sep 2018 12:57:09  WARN PropertyAeFactory - No parameters specified.
22 Sep 2018 12:57:11 ERROR CtakesRestController - Error loading pipers
22 Sep 2018 12:57:11  WARN XmlWebApplicationContext - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'ctakesRestController': Invocation of init method failed; nested exception is javax.servlet.ServletException: org.apache.uima.resource.ResourceInitializationException: EXCEPTION MESSAGE LOCALIZATION FAILED: java.util.MissingResourceException: Can't find resource for bundle java.util.PropertyResourceBundle, key No Analysis Component found for LabValueFinder
22 Sep 2018 12:57:11 ERROR DispatcherServlet - Context initialization failed
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'ctakesRestController': Invocation of init method failed; nested exception is javax.servlet.ServletException: org.apache.uima.resource.ResourceInitializationException: EXCEPTION MESSAGE LOCALIZATION FAILED: java.util.MissingResourceException: Can't find resource for bundle java.util.PropertyResourceBundle, key No Analysis Component found for LabValueFinder
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeInitialization(InitDestroyAnnotationBeanPostProcessor.java:137)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsBeforeInitialization(AbstractAutowireCapableBeanFactory.java:409)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1620)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:555)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:483)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:761)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:867)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:543)
	at org.springframework.web.servlet.FrameworkServlet.configureAndRefreshWebApplicationContext(FrameworkServlet.java:668)
	at org.springframework.web.servlet.FrameworkServlet.createWebApplicationContext(FrameworkServlet.java:634)
	at org.springframework.web.servlet.FrameworkServlet.createWebApplicationContext(FrameworkServlet.java:682)
	at org.springframework.web.servlet.FrameworkServlet.initWebApplicationContext(FrameworkServlet.java:553)
	at org.springframework.web.servlet.FrameworkServlet.initServletBean(FrameworkServlet.java:494)
	at org.springframework.web.servlet.HttpServletBean.init(HttpServletBean.java:171)
	at javax.servlet.GenericServlet.init(GenericServlet.java:158)
	at org.apache.catalina.core.StandardWrapper.initServlet(StandardWrapper.java:1144)
	at org.apache.catalina.core.StandardWrapper.loadServlet(StandardWrapper.java:1091)
	at org.apache.catalina.core.StandardWrapper.allocate(StandardWrapper.java:773)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:133)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:493)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81)
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:650)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:800)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:806)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1498)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:745)
Caused by: javax.servlet.ServletException: org.apache.uima.resource.ResourceInitializationException: EXCEPTION MESSAGE LOCALIZATION FAILED: java.util.MissingResourceException: Can't find resource for bundle java.util.PropertyResourceBundle, key No Analysis Component found for LabValueFinder
	at org.apache.ctakes.rest.service.CtakesRestController$PipelineRunner.<init>(CtakesRestController.java:104)
	at org.apache.ctakes.rest.service.CtakesRestController$PipelineRunner.<init>(CtakesRestController.java:91)
	at org.apache.ctakes.rest.service.CtakesRestController.init(CtakesRestController.java:63)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleElement.invoke(InitDestroyAnnotationBeanPostProcessor.java:366)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleMetadata.invokeInitMethods(InitDestroyAnnotationBeanPostProcessor.java:311)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeInitialization(InitDestroyAnnotationBeanPostProcessor.java:134)
	... 38 more
Caused by: org.apache.uima.resource.ResourceInitializationException: EXCEPTION MESSAGE LOCALIZATION FAILED: java.util.MissingResourceException: Can't find resource for bundle java.util.PropertyResourceBundle, key No Analysis Component found for LabValueFinder
	at org.apache.ctakes.core.pipeline.PiperFileReader.getComponentClass(PiperFileReader.java:285)
	at org.apache.ctakes.core.pipeline.PiperFileReader.addToPipeline(PiperFileReader.java:217)
	at org.apache.ctakes.core.pipeline.PiperFileReader.parsePipelineLine(PiperFileReader.java:164)
	at org.apache.ctakes.core.pipeline.PiperFileReader.loadPipelineFile(PiperFileReader.java:146)
	at org.apache.ctakes.core.pipeline.PiperFileReader.<init>(PiperFileReader.java:114)
	at org.apache.ctakes.rest.service.CtakesRestController$PipelineRunner.<init>(CtakesRestController.java:97)
	... 47 more
```

I think I followed all of the instructions correctly. But, if I missed it then please feel free to call it out.

Thanks!!

**Comment by legendactivated** [comment]
Never mind above error. It was a mistake on my end. I am using mysql 8.0.12 and it didn't allow to have root as a password. So, I chose some other password, but, forgot to update it in custom dictionary. Adding the credentials fixed above error.

After I updated the custom dictionary, I ran into multiple other issues all related to mysql version in pom.xml for ctakes-web-rest and ctakes-codebase-area. 

Errors were:

1. Error creating bean with name 'ctakesRestController': Invocation of init method failed; nested exception is javax.servlet.ServletException: org.apache.uima.resource.ResourceInitializationException: Initialization of annotator class "org.apache.ctakes.dictionary.lookup2.concurrent.ThreadSafeFastLookup" failed.  (Descriptor: <unknown>)

2. java.sql.SQLException: Unable to load authentication plugin 'caching_sha2_password'. 

 I updated mysql version in both pom.xml to the one installed on my machine. But, I got another error:

```
com.mysql.cj.exceptions.InvalidConnectionAttributeException: The server time zone value 'PDT' is unrecognized or represents more than one time zone. You must configure either the server or JDBC driver (via the serverTimezone configuration property) to use a more specifc time zone value if you want to utilize time zone support.
```

which I fixed by adding "&amp;serverTimezone=UTC" to the jdbc url in custom dictionary as below:

```
jdbc:mysql://127.0.0.1:3306/umls?useUnicode=true&amp;characterEncoding=utf8&amp;useSSL=false&amp;serverTimezone=UTC
```

Hopefully this saves someone else from the headache of going through and figuring out what went wrong :)

Thanks!

**Comment by MatthewVi** [comment]
@legendactivated Glad you found help with the new README. Also glad that your problem was resolved.

@gandhirajan Should we add the `&serverTimezone=UTC` by default?

**Comment by lrog** [comment]
Thanks @legendactivated for your comment. I stumbled into the same problem when running MySQL **8.0.12**. I tried the workarounds you mentioned, but still stumbled into more errors (such as: https://stackoverflow.com/questions/49984267/java-sql-sqlexception-unknown-system-variable-query-cache-size ). 
Finally, if it would help anyone, in my case, the quickest solution was just to downgrade the MySQL to **5.7.24** (**5.7** as specified in the README) and everything worked well without any modifications.
